src/Gui/NotificationArea.cpp

```diff
@@ -0,0 +1,993 @@
+    {
+        auto rm = getResourceManager();
+        return rm.error;
+    }
+
+    inline static auto WarningPixmap()
+    {
+        auto rm = getResourceManager();
+        return rm.warning;
+    }
+
+    inline static auto CriticalPixmap()
+    {
+        auto rm = getResourceManager();
+        return rm.critical;
+    }
+
+    inline static auto InfoPixmap()
+    {
+        auto rm = getResourceManager();
+        return rm.info;
+    }
+
+private:
+    QPixmap error;
+    QPixmap warning;
+    QPixmap critical;
+    QPixmap info;
+};
+
+/******************** Console Messages Observer (Console Interface) ************************/
+
+/** This class listens to all messages sent via the console interface and
+   feeds the non-intrusive notification system and the notifications widget */
+class NotificationAreaObserver: public Base::ILogger
+{
+public:
+    NotificationAreaObserver(NotificationArea* notificationarea);
+    ~NotificationAreaObserver() override;
+
+    /// Function that is called by the console interface for this observer with the message
+    /// information
+    void SendLog(const std::string& notifiername, const std::string& msg,
+                 Base::LogStyle level) override;
+
+    /// Name of the observer
+    const char* Name() override
+    {
+        return "NotificationAreaObserver";
+    }
+
+private:
+    NotificationArea* notificationArea;
+};
+
+NotificationAreaObserver::NotificationAreaObserver(NotificationArea* notificationarea)
+    : notificationArea(notificationarea)
+{
+    Base::Console().AttachObserver(this);
+    bLog = false;                  // ignore log messages
+    bMsg = false;                  // ignore messages
+    bNotification = true;          // activate user notifications
+    bTranslatedNotification = true;// activate translated user notifications
+}
+
+NotificationAreaObserver::~NotificationAreaObserver()
+{
+    Base::Console().DetachObserver(this);
+}
+
+void NotificationAreaObserver::SendLog(const std::string& notifiername, const std::string& msg,
+                                       Base::LogStyle level)
+{
+    // 1. As notification system is shared with report view and others, the expectation is that any
+    // individual error and warning message will end in "\n". This means the string must be stripped
+    // of this character for representation in the notification area.
+    // 2. However, any message marked with the QT_TRANSLATE_NOOT macro with the "Notifications"
+    // context, shall not include
+    // "\n", as this generates problems with the translation system. Then the string must be
+    // stripped of "\n" before translation.
+
+    auto simplifiedstring =
+        QString::fromStdString(msg)
+            .trimmed();// remove any leading and trailing whitespace character ('\n')
+
+    // avoid processing empty strings
+    if(simplifiedstring.isEmpty())
+        return;
+
+    if (level == Base::LogStyle::TranslatedNotification) {
```

Comment:
@abdullahtahiriyo 
If level is TranslatedNotification then the text is not translated and otherwise it will be. Shouldn't it be the other way round?

Reply:
OK, I see. Thanks!